File: 2/src/com/reliablecoders/resume/model/ProjectManager.php
<?php
/**
 * Created by PhpStorm.
 * User: pham
 * Date: 2/13/14
 * Time: 3:57 PM
 */

namespace model;
require_once('../dao/Database.php');
require_once('../dao/Project.php');
use dao as dao;

require_once('../dto/ResumeObject.php');
use dto\ResumeObject as ResumeObject;

class ProjectManager
{
    private $database;
    private $project;

    /**
     *
     */
    function __construct()
    {
        $this->database = new dao\Database();
        $this->project = new dao\Project();
    }

    /**
     * @return array
     */
    function selectAllResumes()
    {
        $con = $this->database->getMysqlConnection();
        $result = $this->project->selectAll($con);
        if ($result)
            echo "\xA[" . __CLASS__ . "] Select All Resumes successfully \xA";
        else
            echo "\xA[" . __CLASS__ . "] Select All Resumes ERROR \xA";
        return $result;
    }

    /**
     * @param $id
     * @return array
     */
    function searchResumeByID($id)
    {
        $con = $this->database->getMysqlConnection();
        $result = $this->project->searchResumeByID($con, $id);
        if ($result)
            echo "\xA[" . __CLASS__ . "] Search Resume by ID \"" . $id . "\" successfully \xA";
        else
            echo "\xA[" . __CLASS__ . "] Search Resume by ID \"" . $id . "\" ERROR \xA";
        return $result[0];
    }

    /**
     * @param $value
     * @return array
     */
    function searchResumeByValue($value)
    {
        $con = $this->database->getMysqlConnection();
        $result = $this->project->searchResumeByValue($con, $value);
        if ($result)
            echo "\xA[" . __CLASS__ . "] Search Resume by value \"" . $value . "\" successfully \xA";
        else
            echo "\xA[" . __CLASS__ . "] Search Resume by value \"" . $value . "\" ERROR \xA";
        return $result;
    }

    /**
     * @param $resumeObj
     * @return ResumeObject
     */
    public function addResume($resumeObj)
    {
        $con = $this->database->getMysqlConnection();
        $result = $this->project->addResume($con, $resumeObj);
        if ($result)
            echo "\xA[" . __CLASS__ . "] Add new resume successfully \xA";
        else
            echo "\xA[" . __CLASS__ . "] Add new resume ERROR \xA";
        return $result;
    }

    public function updateResume($resumeObj)
    {
        $con = $this->database->getMysqlConnection();
        $result = $this->project->updateResumeByID($con, $resumeObj);
        if ($result)
            echo "\xA[" . __CLASS__ . "] Update resume successfully \xA";
        else
            echo "\xA[" . __CLASS__ . "] Update resume ERROR \xA";
        return $result;
    }
    public function deleteResume($resumeObj)
    {
        $con = $this->database->getMysqlConnection();
        $result = $this->project->deleteResumeByID($con, $resumeObj);
        if ($result)
            echo "\xA[" . __CLASS__ . "] Delete resume successfully \xA";
        else
            echo "\xA[" . __CLASS__ . "] Delete resume ERROR \xA";
        return $result;
    }
}

File: 2/src/com/reliablecoders/resume/model/ProjectManager.java
package com.reliablecoders.resume.model;

import java.sql.Connection;
import java.util.ArrayList;
import java.util.List;

import com.reliablecoders.resume.dao.Database;
import com.reliablecoders.resume.dao.Project;
import com.reliablecoders.resume.dto.ResumeObject;
/**
 * @author nam phan
 * @company Reliable{coders}
 */
public class ProjectManager {

	/**
	 * 
	 * @return
	 * @throws Exception
	 */
	public ArrayList<ResumeObject> GetResumes() throws Exception {
		ArrayList<ResumeObject> resumes = null;
		try {
			Database database = new Database();
			Connection connection = database.Get_Connection();
			Project project = new Project();
			resumes = project.GetAllResumes(connection);

		} catch (Exception e) {
			throw e;
		}
		return resumes;
	}

	/**
	 * 
	 * @param resumeObject
	 * @throws Exception
	 */
	public ResumeObject AddResume(ResumeObject resumeObject) throws Exception {
		try {
			Database database = new Database();
			Connection connection = database.Get_Connection();
			Project project = new Project();
			return project.AddResume(resumeObject, connection);
		} catch (Exception e) {
			throw e;
		}

	}
	/**
	 * 
	 * @param value
	 * @return
	 * @throws Exception 
	 */
	public ArrayList<ResumeObject> SearchResumes(String value) throws Exception {
		ArrayList<ResumeObject> resumes = null;
		try {
			Database database = new Database();
			Connection connection = database.Get_Connection();
			Project project = new Project();
			resumes = project.SearchResumes(connection, value);

		} catch (Exception e) {
			throw e;
		}
		return resumes;
	}
	/**
	 * 
	 * @param resumeObject
	 * @return
	 * @throws Exception 
	 */
	public ResumeObject updateResume(ResumeObject resumeObject) throws Exception {
		try {
			Database database = new Database();
			Connection connection = database.Get_Connection();
			Project project = new Project();
			return project.UpdateResume(resumeObject, connection);
		} catch (Exception e) {
			throw e;
		}
		
	}
	/**
	 * 
	 * @return
	 * @throws Exception 
	 */
	public int getLastInsertedID() throws Exception {
		Database database = new Database();
		Connection connection = database.Get_Connection();
		Project project = new Project();
		return project.getLastInsertedID(connection);
		
	}
	/**
	 * 
	 * @param lastID
	 * @return
	 * @throws Exception 
	 */
	public ResumeObject SearchResumesByID(int lastID) throws Exception {
		Database database = new Database();
		Connection connection = database.Get_Connection();
		Project project = new Project();
		return project.SearchResumeByID(connection, lastID);	}
	/**
	 * 
	 * @param resumes
	 * @return
	 * @throws Exception
	 */
	public List<ResumeObject> deleteResumes(List<ResumeObject> resumes) throws Exception {
		
		ArrayList<ResumeObject> array = new ArrayList<ResumeObject>();
		for(ResumeObject resume : resumes)
		{
			array.add(this.deleteResume(resume));
		}
		return array;
		
	}
	/**
	 * 
	 * @param resume
	 * @return
	 * @throws Exception
	 */
	private ResumeObject deleteResume(ResumeObject resume) throws Exception {
		Database database = new Database();
		Connection connection = database.Get_Connection();
		Project project = new Project();
		return project.DeleteResumeByID(connection, resume);
	}
	/**
	 * 
	 * @param value
	 * @return
	 * @throws Exception 
	 * @throws NumberFormatException 
	 */
	public ResumeObject SearchByID(String id) throws NumberFormatException, Exception {
		Database database = new Database();
		Connection connection = database.Get_Connection();
		Project project = new Project();
		return project.SearchResumeByID(connection, Integer.parseInt(id));
	}

}
